src/sentry/web/api.py

```diff
@@ -390,162 +547,60 @@ def process(self, request, project, key, auth, helper, data, attachments=None, *
-                sender=type(self),
-            )
-            raise APIForbidden('Event dropped due to filter: %s' % (filter_reason,))
+        self.pre_normalize(event_manager, helper)
+        event_manager.normalize()
 
-        # TODO: improve this API (e.g. make RateLimit act on __ne__)
-        rate_limit = safe_execute(
-            quotas.is_rate_limited, project=project, key=key, _with_transaction=False
-        )
-        if isinstance(rate_limit, bool):
-            rate_limit = RateLimit(is_limited=rate_limit, retry_after=None)
-
-        # XXX(dcramer): when the rate limiter fails we drop events to ensure
-        # it cannot cascade
-        if rate_limit is None or rate_limit.is_limited:
-            if rate_limit is None:
-                api_logger.debug('Dropped event due to error with rate limiter')
-            tsdb.incr_multi(
-                [
-                    (tsdb.models.project_total_received, project.id),
-                    (tsdb.models.project_total_rejected, project.id),
-                    (tsdb.models.organization_total_received,
-                     project.organization_id),
-                    (tsdb.models.organization_total_rejected,
-                     project.organization_id),
-                    (tsdb.models.key_total_received, key.id),
-                    (tsdb.models.key_total_rejected, key.id),
-                ],
-                timestamp=tsdb_start_time,
-            )
-            metrics.incr(
-                'events.dropped',
-                tags={
-                    'reason': rate_limit.reason_code if rate_limit else 'unknown',
-                }
-            )
-            event_dropped.send_robust(
-                ip=remote_addr,
-                project=project,
-                sender=type(self),
-                reason_code=rate_limit.reason_code if rate_limit else None,
-            )
-            if rate_limit is not None:
-                raise APIRateLimited(rate_limit.retry_after)
-        else:
-            tsdb.incr_multi(
-                [
-                    (tsdb.models.project_total_received, project.id),
-                    (tsdb.models.organization_total_received,
-                     project.organization_id),
-                    (tsdb.models.key_total_received, key.id),
-                ],
-                timestamp=tsdb_start_time,
-            )
+        event_type = type(self).type_name
 
-        org_options = OrganizationOption.objects.get_all_values(
-            project.organization_id)
+        # TODO: Some form of coordination between the Kafka consumer
+        # and this method (the 'relay') to decide whether a 429 should
+        # be returned here.
 
-        data = event_mgr.get_data()
-        del event_mgr
+        # Everything before this will eventually be done in the relay.
+        if (kafka_publisher is not None
+                and not attachments
+                and random.random() < options.get('store.kafka-sample-rate')):
```

Comment:
RE: `and not attachments` I wanted to make this a followup PR. I hear we limit attachments to ~50MB or something in Nginx, we'll need to figure out if that's a total per event or per attachment, and increase the max message size in Kafka. Then we'll need to gzip+base64 (I assume) the attachments into the JSON payload.